File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/mangers/live_room_event_sender_manger.dart
import 'dart:convert';

import 'package:flutter/foundation.dart';
import 'package:matlive_client_sdk_flutter/audio/define/events.dart';
import 'package:matlive_client_sdk_flutter/audio/mangers/mat_live_room_manger.dart';
import 'package:matlive_client_sdk_flutter/audio/mangers/mat_live_join_room_manger.dart';

class LiveRoomEventSenderManger {
  Future<void> _publish({required Map<String, dynamic> data}) async {
    final room = MatLiveRoomManger.instance.room;
    if (room != null) {
      data['user'] = {
        'userId': MatLiveJoinRoomManger.instance.currentUser?.userId,
        'name': MatLiveJoinRoomManger.instance.currentUser?.name,
        'avatar': MatLiveJoinRoomManger.instance.currentUser?.avatar,
      };
      data['roomId'] = room.name;
      Uint8List decoded = utf8.encode(jsonEncode(data));
      await room.localParticipant?.publishData(decoded, reliable: false);
    }
  }

  Future<void> sendMessage(String message) async {
    await _publish(data: {
      'event': MatLiveEvents.sendMessage,
      'message': message,
    });
  }

  Future<void> sendGift(String gift) async {
    await _publish(data: {
      'event': MatLiveEvents.sendGift,
      'gift': gift,
    });
  }

  Future<void> clearChat() async {
    await _publish(data: {
      'event': MatLiveEvents.clearChat,
    });
  }

  Future<void> inviteUserToTakeMic(String userId, int seatIndex) async {
    await _publish(data: {
      'event': MatLiveEvents.inviteUserToTakeMic,
      'seatIndex': seatIndex,
      'userId': userId,
    });
  }

  Future<void> requestTakeMic(int seatIndex) async {
    await _publish(data: {
      'event': MatLiveEvents.requestTakeMic,
      'seatIndex': seatIndex,
    });
  }
}

File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/mangers/live_room_event_sender_manger.kt
//package com.matnsolutions.matlive_sdk.audio.mangers
//
//import com.matnsolutions.matlive_sdk.audio.define.MatLiveEvents
//import org.json.JSONObject
//
//import java.nio.charset.StandardCharsets
//
//interface LiveRoomEventSenderManger {
//
//    private suspend fun publish(data: Map<String, Any>) {
//        val room = MatLiveRoomManger.instance.room
//        if (room != null) {
//            val jsonObject = JSONObject(data)
//
//            // Add user data
//            val userObject = JSONObject().apply {
//                put("userId", MatLiveJoinRoomManger.instance.currentUser?.userId)
//                put("name", MatLiveJoinRoomManger.instance.currentUser?.name)
//                put("avatar", MatLiveJoinRoomManger.instance.currentUser?.avatar)
//            }
//            jsonObject.put("user", userObject)
//
//            // Add room ID
//            jsonObject.put("roomId", room.name.toString())
//
//            // Convert to byte array and publish
//            val decoded = jsonObject.toString().toByteArray(StandardCharsets.UTF_8)
//            room.localParticipant.publishData(decoded)
//        }
//    }
//
//     suspend fun sendMessage(message: String) {
//        publish(
//            mapOf(
//                "event" to MatLiveEvents.sendMessage,
//                "message" to message
//            )
//        )
//    }
//
//    suspend fun sendGift(gift: String) {
//        publish(
//            mapOf(
//                "event" to MatLiveEvents.sendGift,
//                "gift" to gift
//            )
//        )
//    }
//
//    suspend fun clearChat() {
//        publish(
//            mapOf(
//                "event" to MatLiveEvents.clearChat
//            )
//        )
//    }
//
//    suspend fun inviteUserToTakeMic(userId: String, seatIndex: Int) {
//        publish(
//            mapOf(
//                "event" to MatLiveEvents.inviteUserToTakeMic,
//                "seatIndex" to seatIndex,
//                "userId" to userId
//            )
//        )
//    }
//
//    suspend fun requestTakeMic(seatIndex: Int) {
//        publish(
//            mapOf(
//                "event" to MatLiveEvents.requestTakeMic,
//                "seatIndex" to seatIndex
//            )
//        )
//    }
//}